.github/workflows/macos-build.yml

```diff
@@ -3,41 +3,60 @@ name: Build MacOS
 # TODO: change to 'pull_request' and 'push' 
 # when we are ready to enable this for PRs.
 on:
-  workflow_dispatch:
+  workflow_call:
+    inputs:
+      IS_RELEASE:
+        required: true
+        type: boolean
+        default: false
+      GN_CONFIG:
+        required: false
+        type: string
+        default: false//electron/build/args/testing.gn
+      GN_BUILD_TYPE:
+        required: false
+        type: string
+        default: testing
+      GENERATE_SYMBOLS: 
+        required: false
+        type: boolean
+        default: false
+      UPLOAD_TO_STORAGE: 
+        required: false
+        type: string
+        default: '0'
+      
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
 env:
-  GCLIENT_EXTRA_ARGS: '--custom-var=checkout_mac=True --custom-var=host_os=mac --custom-var=host_cpu=arm64'
```

Comment:
I thought custom-var=host_cpu=arm64 was needed to make sure we get everything we need for arm builds

Reply:
@jkleinsc it is - this was from the initial hardcoding without cross-compilation i think. It looks like this got partially refactored down to https://github.com/electron/electron/pull/42236/files#diff-3569b5feda562f1b9cb8c965c02dc80df7a1969e39e0a8ecb0439553a248222aR219-R230, but i'm not finding where that gets dynamically then used in extra args 🤔 @VerteDinde where is that done?

Reply:
Leaving a comment here to pull a conversation out of Slack - we did originally add this back in, but passing this into `GCLIENT_EXTRA_ARGS ` invalidates the former DEPSHASH cache number, which means we'll need to checkout the cache twice. We'll test the full flow while I hit that in a follow up PR